core/templates/dev/head/pages/exploration_player/AnswerClassificationService.js

```diff
@@ -125,27 +118,42 @@ oppia.factory('AnswerClassificationService', [
           return deferred.promise;
         }
 
-        if (result.outcome === defaultOutcome &&
-            INTERACTION_SPECS[oldState.interaction.id]
-              .is_string_classifier_trainable &&
-            ENABLE_ML_CLASSIFIERS) {
-          var classifyUrl = '/explorehandler/classify/' + explorationId;
-          var params = (
-            isInEditorMode ? {} : LearnerParamsService.getAllParams());
+        var ruleBasedOutcomeIsDefault = (result.outcome === defaultOutcome);
+        var interactionIsTrainable = INTERACTION_SPECS[
+          oldState.interaction.id].is_interaction_trainable;
 
-          $http.post(classifyUrl, {
-            old_state: oldState.toBackendDict(),
-            params: params,
-            answer: answer
-          }).then(function(response) {
-            var result = response.data;
-            deferred.resolve({
-              outcome: result.outcome,
-              ruleIndex: result.rule_spec_index,
-              answerGroupIndex: result.answer_group_index,
-              classificationCategorization: result.classification_categorization
-            });
-          });
+        if (ruleBasedOutcomeIsDefault && interactionIsTrainable &&
+            ENABLE_ML_CLASSIFIERS) {
+          var classifierData = StateClassifierMappingService.getClassifierData(
+            stateName);
+          var algorithmId = StateClassifierMappingService.getAlgorithmId(
+            stateName);
+          var dataSchemaVersion = (
+            StateClassifierMappingService.getDataSchemaVersion(stateName));
+          if (classifierData && algorithmId && dataSchemaVersion) {
+            var predictionService = (
+              PredictionAlgorithmRegistryService.getPredictionService(
+                algorithmId, dataSchemaVersion));
+            var predictedAnswerGroupIndex;
+            // If prediction service exists, we run classifier. We return the
+            // default outcome otherwise.
+            if (predictionService) {
+              predictedAnswerGroupIndex = predictionService.predict(
+                classifierData, answer);
+              var classificationResult = AnswerClassificationResult.createNew(
+                answerGroups[predictedAnswerGroupIndex].outcome,
+                predictedAnswerGroupIndex,
+                findRuleIndex(answerGroups[predictedAnswerGroupIndex],
+                              predictedAnswerGroupIndex),
+                TRAINING_DATA_CLASSIFICATION
+              );
+            } else {
+              var classificationResult = result;
```

Comment:
Seems weird to give it this name, especially since classification didn't happen...

Why not assign the thing on line 143 to the `result` variable, get rid of this else block, get rid of the other else block below and just do `deferred.resolve(result);` at the end?